src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,444 @@
+
+                    // input
+                    final byte[] source = eventToSource();
+
+                    // origin
+                    final byte[] origin = eventToOrigin();
+
+                    // Format: Use AVRO format with syslog columns as indexed ones
+                    final long epochMicros = rfc3339ToEpoch(
+                            new RFC5424Timestamp(rfc5424Frame.timestamp).toZonedDateTime()
+                    );
+                    SyslogRecord syslogRecord = SyslogRecord
+                            .newBuilder()
+                            .setTimestamp(epochMicros)
+                            .setPayload(rfc5424Frame.msg.toString())
+                            .setDirectory(rfc5424Frame.structuredData.getValue(teragrepDirectory).toString())
+                            .setStream(rfc5424Frame.structuredData.getValue(teragrepStreamName).toString()) // Or is sourcetype/stream supposed to be rfc5424Frame.appName.toString() instead?
+                            .setHost(rfc5424Frame.hostname.toString())
+                            .setInput(new String(source, StandardCharsets.UTF_8))
+                            .setPartition(recordOffsetObjectJo.get("partition").getAsString())
+                            .setOffset(recordOffsetObjectJo.get("offset").getAsLong())
+                            .setOrigin(new String(origin, StandardCharsets.UTF_8))
+                            .build();
+
+                    // Calculate the size of syslogRecord that is going to be written to syslogAvroWriter-file.
+                    long capacity = syslogRecord.toByteBuffer().capacity();
+                    // Check if there is still room in syslogAvroWriter for another syslogRecord. Commit syslogAvroWriter to HDFS if no room left, emptying it out in the process.
+                    boolean fileCommitted = writeToHdfs(syslogAvroWriter.getFileSize() + capacity, lastObjectJo);
+                    if (fileCommitted) {
+                        // This part defines a new empty file to which the new AVRO-serialized records are stored until it again hits the size limit defined in config.
+                        writableQueue
+                                .setQueueNamePrefix(recordOffsetObjectJo.get("topic").getAsString() + recordOffsetObjectJo.get("partition").getAsString());
+                        syslogFile = writableQueue.getNextWritableFile();
+                        syslogAvroWriter = new SyslogAvroWriter(syslogFile);
+                        if (LOGGER.isDebugEnabled()) {
+                            LOGGER
+                                    .debug(
+                                            "Target file size reached, file <{}> stored to <{}> in HDFS", syslogFile
+                                                    .getName(),
+                                            lastObjectJo.get("topic").getAsString() + "/" + lastObjectJo.get("partition").getAsString() + "." + lastObjectJo.get("offset").getAsString()
```

Comment:
I'd rather pass the last argument as multiple arguments and have `<{}/{}.{}>` in the template

Reply:
Solved in commit 06497539abcc630a07e71a2d28b19e1e0c2a1dc3